### PR TITLE
Jupyter notebook improvements for Hackathon 2 training

### PR DESCRIPTION
**Pull request by rcalvert-cpi**
In order to provide more useful VOLTTRON training materials (and to support the second Hackathon at SLAC), enhance the notebooks in examples/jupyter_notebooks as follows:

Allow one notebook to import code from another notebook.
Support the ability to insert Python code from a notebook into a VOLTTRON agent's source code.
Provide an example of using the OpenADRVenAgent.
Also, in order to use the latest version of the jupyter library, relax the constraint that has been preventing VOLTTRON from using newer versions of the pyzmq library.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/VOLTTRON/volttron/pull/1805%23issuecomment-423324397%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1805%23discussion_r219646794%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1805%23discussion_r219660957%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1805%23discussion_r219660990%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1805%23issuecomment-423324397%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Issue%20%231804%22%2C%20%22created_at%22%3A%20%222018-09-20T20%3A36%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/11467133%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rcalvert-cpi%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20def99efe4a289b9af425d8924300333397f92855%20services/core/OpenADRVenAgent/requirements.txt%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1805%23discussion_r219646794%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20are%20these%20no%20longer%20required%3F%22%2C%20%22created_at%22%3A%20%222018-09-21T23%3A01%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/3979063%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/craig8%22%7D%7D%2C%20%7B%22body%22%3A%20%22That%20file%20was%20deleted%20in%20error.%20Thanks%20for%20catching%20the%20mistake%21%20I%27ll%20put%20it%20back.%22%2C%20%22created_at%22%3A%20%222018-09-22T04%3A14%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/11467133%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rcalvert-cpi%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/OpenADRVenAgent/requirements.txt%3AL0-0%22%7D%2C%20%22Pull%20def99efe4a289b9af425d8924300333397f92855%20requirements.txt%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1805%23discussion_r219660957%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Yes%2C%20Jupyter%20now%20requires%20a%20fairly%20current%20version%20of%20pyzmq.%20We%20had%20a%20quick%20conversation%20about%20it%20a%20month%20or%20so%20ago.%20You%20thought%20that%20VOLTTRON%27s%20restriction%20to%20an%20older%20version%20of%20pyzmq%20was%20probably%20for%20reasons%20that%20are%20now%20obsolete.%22%2C%20%22created_at%22%3A%20%222018-09-22T04%3A12%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/11467133%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rcalvert-cpi%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20requirements.txt%3AL11-18%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1805#issuecomment-423324397'>General Comment</a></b>
- <a href='https://github.com/rcalvert-cpi'></a> Issue #1804
- [ ] <a href='#crh-comment-Pull def99efe4a289b9af425d8924300333397f92855 services/core/OpenADRVenAgent/requirements.txt 1'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1805#discussion_r219646794'>File: services/core/OpenADRVenAgent/requirements.txt:L0-0</a></b>
- <a href='https://github.com/craig8'></a> Why are these no longer required?
- <a href='https://github.com/rcalvert-cpi'></a> That file was deleted in error. Thanks for catching the mistake! I'll put it back.
- [ ] <a href='#crh-comment-Pull def99efe4a289b9af425d8924300333397f92855 requirements.txt 4'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1805#discussion_r219660957'>File: requirements.txt:L11-18</a></b>
- <a href='https://github.com/rcalvert-cpi'></a> Yes, Jupyter now requires a fairly current version of pyzmq. We had a quick conversation about it a month or so ago. You thought that VOLTTRON's restriction to an older version of pyzmq was probably for reasons that are now obsolete.

<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1805?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1805?approve=1'></a>&nbsp;<a href='https://github.com/VOLTTRON/volttron/pull/1805'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by craig8** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/VOLTTRON/volttron/pull/1805'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by rcalvert-cpi** [comment]
Issue #1804

File: requirements.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ pymodbus==1.2.0
 pyserial==3.1.1
 python-dateutil==2.5.3
 pytz==2016.6.1
-pyzmq==15.4.0
```
Comment (craig8) [inline requirements.txt]: I assume there was a need to upgrade this?

File: services/core/OpenADRVenAgent/requirements.txt [hunk 2]
```diff
@@ -1,4 +0,0 @@
-isodate==0.6
```
Comment (craig8) [inline services/core/OpenADRVenAgent/requirements.txt]: Why are these no longer required?

File: requirements.txt (hunk 1 above)
Comment (rcalvert-cpi) [reply requirements.txt]: Yes, Jupyter now requires a fairly current version of pyzmq. We had a quick conversation about it a month or so ago. You thought that VOLTTRON's restriction to an older version of pyzmq was probably for reasons that are now obsolete.

File: services/core/OpenADRVenAgent/requirements.txt (hunk 2 above)
Comment (rcalvert-cpi) [reply services/core/OpenADRVenAgent/requirements.txt]: That file was deleted in error. Thanks for catching the mistake! I'll put it back.

**Pull request closed without merging**